dissect/target/plugins/child/wsl.py

```diff
@@ -7,33 +7,37 @@
 
 
 def find_wsl_installs(target: Target) -> Iterator[Path]:
-    # Officially supported distro's by Microsoft can be found under "PackageFamilyName" at
-    # https://github.com/microsoft/WSL/blob/master/distributions/DistributionInfo.json
+    """Find all WSL disk files.
 
-    dist_folders = [
-        "CanonicalGroupLimited.Ubuntu*",
-        "TheDebianProject.DebianGNULinux_*",
-        "KaliLinux.*",
-        "*SUSE.openSUSE*",
-        "*OracleAmericaInc.OracleLinux*",
-    ]
+    Disk files for working (custom) Linux distributions can be located anywhere on the system.
+    Locations to disk files for each user's WSL instance is stored in the Windows registry at
+    ``HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Lxss``.
 
-    for user_details in target.user_details.all_with_home():
-        for dist_folder in dist_folders:
-            for install_path in user_details.home_path.joinpath("AppData/Local/Packages").glob(dist_folder):
-                if (vhdx := install_path.joinpath("LocalState/ext4.vhdx")).exists():
-                    yield vhdx
+    References:
+        - https://learn.microsoft.com/en-us/windows/wsl/use-custom-distro
+        - https://learn.microsoft.com/en-us/windows/wsl/enterprise
+    """
+
+    for lxss_key in target.registry.keys("HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Lxss"):
+        for distribution_key in lxss_key.subkeys():
+            if not distribution_key.name.startswith("{"):
+                continue
+            base_path = distribution_key.value("BasePath").value.replace(
```

Comment:
My only comment is that this should be properly fixed, otherwise this is good to go!